hpx/util/cache_aligned_data.hpp

```diff
@@ -30,7 +30,8 @@ namespace hpx { namespace util
 
         // pad to cache line size bytes
         Data data_;
-        char cacheline_pad[threads::get_cache_line_size() - sizeof(Data)];
+        char cacheline_pad[threads::get_cache_line_size() -
+            (sizeof(Data) % threads::get_cache_line_size())];
     };
```

Comment:
We'll need to adapt the `static_assert` above in this case.

Reply:
Indeed, good catch. Thanks!

Reply:
Indeed, good catch. Thanks!